common/fee/models.go

```diff
@@ -0,0 +1,183 @@
+
+}
+
+func GetFeeCap(cfg feetypes.FixedPriceEstimatorConfig, originalFeeLimit uint32, maxFeePrice *big.Int) (feeCap *big.Int) {
+	if cfg.BumpThreshold() == 0 {
+		// Fee bumping is disabled, just use the max fee cap
+		feeCap = getMaxFeePrice(maxFeePrice, cfg.PriceMax())
+	} else {
+		// Need to leave headroom for bumping so we fallback to the default value here
+		feeCap = cfg.FeeCapDefault()
+	}
+	return feeCap
+}
+
+// BumpDynamicFeeOnly bumps the tip cap and max fee price if necessary
+func CalculateBumpDynamicFee(cfg feetypes.BumpConfig, feeCapBufferBlocks uint16, lggr logger.SugaredLogger, currentTipCap, currentBaseFee *big.Int, originalFeeCap, originalTipCap *big.Int, originalFeeLimit uint32, maxFeePrice *big.Int) (bumpedFeeCap, bumpedTipCap *big.Int, chainSpecificFeeLimit uint32, err error) {
+	bumpedFeeCap, bumpedTipCap, err = bumpDynamicFee(cfg, feeCapBufferBlocks, lggr, currentTipCap, currentBaseFee, originalFeeCap, originalTipCap, maxFeePrice)
+	if err != nil {
+		return bumpedFeeCap, bumpedTipCap, 0, err
+	}
+	chainSpecificFeeLimit = ApplyMultiplier(originalFeeLimit, cfg.LimitMultiplier())
+	return
+}
+
+func bumpDynamicFee(cfg feetypes.BumpConfig, feeCapBufferBlocks uint16, lggr logger.SugaredLogger, currentTipCap, currentBaseFee *big.Int, originalFeeCap, originalTipCap *big.Int, maxFeePriceInput *big.Int) (bumpedFeeCap, bumpedTipCap *big.Int, err error) {
+	maxFeePrice := getMaxFeePrice(maxFeePriceInput, cfg.PriceMax())
+	baselineTipCap := max(originalTipCap, cfg.TipCapDefault())
+	bumpedTipCap = bumpFeePriceByPercentage(baselineTipCap, cfg.BumpPercent(), cfg.BumpMin())
+
+	// Update bumpedTipCap if currentTipCap is higher than bumpedTipCap and within maxFeePrice
+	bumpedTipCap = maxBumpedFee(lggr, currentTipCap, bumpedTipCap, maxFeePrice, "tip cap")
+
+	if bumpedTipCap.Cmp(maxFeePrice) > 0 {
+		return bumpedFeeCap, bumpedTipCap, errors.Wrapf(ErrBumpFeeExceedsLimit, "bumped tip cap of %s would exceed configured max fee price of %s (original fee: tip cap %s, fee cap %s). %s",
+			bumpedTipCap.String(), maxFeePrice, originalTipCap.String(), originalFeeCap.String(), label.NodeConnectivityProblemWarning)
+	} else if bumpedTipCap.Cmp(originalTipCap) <= 0 {
+		// NOTE: This really shouldn't happen since we enforce minimums for
+		// FeeEstimator.BumpPercent and FeeEstimator.BumpMin in the config validation,
+		// It's here for extra precaution
+		return bumpedFeeCap, bumpedTipCap, errors.Wrapf(ErrBump, "bumped fee tip cap of %s fee unit is less than or equal to original fee tip cap of %s fee unit."+
+			" ACTION REQUIRED: This is a configuration error, you must increase either "+
+			"FeeEstimator.BumpPercent or FeeEstimator.BumpMin", bumpedTipCap.String(), originalTipCap)
+	}
+
+	// Always bump the FeeCap by at least the bump percentage
+	// For geth the configured bump is 10%
+	bumpedFeeCap = max(
+		addPercentage(originalFeeCap, cfg.BumpPercent()),
+		new(big.Int).Add(originalFeeCap, cfg.BumpMin()),
+	)
```

Comment:
Is this different math than `bumpFeePriceByPercentage`?

Reply:
thanks for the note, we can refactor them into `bumpFeePriceByPercentage`